tests/ext/httplib/test_httplib.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+import requests
+
+from aws_xray_sdk.core import patch
+from aws_xray_sdk.core import xray_recorder
+from aws_xray_sdk.core.context import Context
+
+
+patch(('httplib',))
+
+# httpbin.org is created by the same author of requests to make testing http easy.
+BASE_URL = 'httpbin.org'
+
+
+@pytest.fixture(autouse=True)
+def construct_ctx():
+    """
+    Clean up context storage on each test run and begin a segment
+    so that later subsegment can be attached. After each test run
+    it cleans up context storage again.
+    """
+    xray_recorder.configure(service='test', sampling=False, context=Context())
+    xray_recorder.clear_trace_entities()
+    xray_recorder.begin_segment('name')
+    yield
+    xray_recorder.clear_trace_entities()
+
+
+def test_ok():
+    status_code = 200
+    url = 'http://{}/status/{}?foo=bar&baz=foo'.format(BASE_URL, status_code)
+    requests.get(url)
+    subsegment = xray_recorder.current_segment().subsegments[1]
+    assert subsegment.name == url
```

Comment:
Can this assertion pass when query string is stripped?

Reply:
yes, the existing code just prints a logging warning, so this helps at least to see it in the logs.

Reply:
sorry, bug

Reply:
Based on your patching code you use the string before "?" as the subsegment name.
So to verify the http subsegment is named correctly, should the test asserts name == `http://{}/status/{}` and pass?

Reply:
fixed